### PR TITLE
Missing functionality

### PR DESCRIPTION
**Pull request by rsoeldner**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,22 +1,20 @@
-name := "gecko"
 
-version := "0.1"
+lazy val deps = libraryDependencies := {
+  val catsV       = "1.0.0-MF"
```
Comment (jmcardon) [inline build.sbt]: Cats is on RC1 now.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,16 @@
 # gecko
 array-backed and _predictable_ data manipulation library inspired by _saddle_ and _pandas_.
 
+## dependency
+We rely on 
+
+| Name     | Version   |
+| ---      | ---       |
+| cats     | 1.0.0-MF  |
```
Comment (jmcardon) [inline README.md]: Should probably use RC1 to start, then make sure we update when cats releases 1.0.0

File: src/main/scala/gecko/csv/GeckoCSVUtil.scala [hunk 3]
```diff
@@ -49,6 +49,44 @@ object GeckoCSVUtil {
         .covary[F]
   }
 
+  def streamWithRowIx[F[_]: Sync, R, C, A: ClassTag](
```
Comment (jmcardon) [inline src/main/scala/gecko/csv/GeckoCSVUtil.scala]: This needs tests (From me)

File: src/main/scala/gecko/csv/GeckoCSVUtil.scala [hunk 4]
```diff
@@ -14,12 +14,12 @@ object GeckoCSVUtil {
   private val quoteChar = '"'
   private val separator = ','
 
-  def streamFrame[F[_] : Sync, R, C, A: ClassTag](
-                                                   frame: DataFrame[R, C, A]
```
Comment (jmcardon) [inline src/main/scala/gecko/csv/GeckoCSVUtil.scala]: you know, it's diffs like this in both this project and `tsec` which make me think we really need to maybe start using scalafmt @ the root, then ensure that we run it before every big merge.

It's kind of silly this happens.

File: src/main/scala/InternalBenchmarks.scala [hunk 5]
```diff
@@ -1,111 +0,0 @@
...
-      arr
-    }
-
-  }
-
-  def main(args: Array[String]): Unit = {
-
-    val th = Thyme.warmed(verbose = print)
-
-    //    val arrayGString1: Array[String] =
```
Comment (jmcardon) [inline src/main/scala/InternalBenchmarks.scala]: Let's write some sbt-jmh benchmarks so our actual performance is brought to light, no?

**Comment by jmcardon** [comment]
Also another note: `mapCol` actually is applying `f` without checking emptyness of values. If our property tests didn't catch this, is because we're failing to generate frames with `Empty` values for object types. This _would_ throw an NPE.

File: README.md (hunk 2 above)
Comment (rsoeldner) [reply README.md]: good point :+1:

Comment (rsoeldner) [reply README.md]: @jmcardon, yankDistinct is called groupBy, we have it already

**Comment by rsoeldner** [comment]
Whoop, we had a small bug
```scala
val add1: Int => Int = (v: Int) => v + 1
      val res: DataVector[Int] = DataVector(1,2,3, geckoInt.emptyElement).map(add1)
      res.underlying should be(DataVector(2,3,4, Int.MinValue).underlying)
```
mapped over the empty element. This was because of an implicit factory for `emptyGecko`.

**Pull request merged**